tests/UNMOCKED_end2end_tests/run_all_unmocked_tests.sh

```diff
@@ -3,44 +3,48 @@ set -e
 
 # Get the parameters
 export TOKEN=$1
-echo "token: ${TOKEN}"
+[[ -z "$TOKEN" ]] && { echo "Error: token is empty" ; exit 1; }
+echo "############################### token: ${TOKEN}"
 
 echo "############################### Download the clothing_dataset_small"
 
 DIR_DATASET=clothing_dataset_small
-if [ -d "DIR_DATASET" ]; then
-  echo "Skipping download of dataset, it already exists"
+if [ -d $DIR_DATASET ]; then
+  echo "Skipping download of dataset, it already exists."
 else
-  git clone https://github.com/alexeygrigorev/clothing-dataset-small DIR_DATASET
+  git clone https://github.com/alexeygrigorev/clothing-dataset-small $DIR_DATASET
 fi
 INPUT_DIR="${DIR_DATASET}/test/dress"
 
-
 # Run the tests
 echo "############################### Test 1"
-lightly-magic token=${TOKEN} input_dir=${INPUT_DIR} trainer.max_epochs=0 new_dataset_name=test_unmocked_cli_1
+lightly-magic token=$TOKEN input_dir=$INPUT_DIR trainer.max_epochs=0 new_dataset_name=test_unmocked_cli_1
 
 echo "############################### Test 2"
-lightly-magic token=${TOKEN} input_dir=${INPUT_DIR} trainer.max_epochs=1 new_dataset_name=test_unmocked_cli_2
+lightly-magic token=$TOKEN input_dir=$INPUT_DIR trainer.max_epochs=1 new_dataset_name=test_unmocked_cli_2
 
 echo "############################### Test 3"
-lightly-upload token=${TOKEN} input_dir=${INPUT_DIR} new_dataset_name=test_unmocked_cli_3
-lightly-upload token=${TOKEN} input_dir=${INPUT_DIR} new_dataset_name=test_unmocked_cli_3
+lightly-upload token=$TOKEN input_dir=$INPUT_DIR new_dataset_name=test_unmocked_cli_3
+lightly-upload token=$TOKEN input_dir=$INPUT_DIR new_dataset_name=test_unmocked_cli_3
 
 echo "############################### Test 4"
-lightly-magic token=${TOKEN} input_dir=${INPUT_DIR} trainer.max_epochs=0 new_dataset_name=test_unmocked_cli_4
-lightly-magic token=${TOKEN} input_dir=${INPUT_DIR} trainer.max_epochs=0 new_dataset_name=test_unmocked_cli_4
+lightly-magic token=$TOKEN input_dir=$INPUT_DIR trainer.max_epochs=0 new_dataset_name=test_unmocked_cli_4
+lightly-magic token=$TOKEN input_dir=$INPUT_DIR trainer.max_epochs=0 new_dataset_name=test_unmocked_cli_4
 
 echo "############################### Test 5"
-lightly-upload token=${TOKEN} input_dir=${INPUT_DIR} new_dataset_name=test_unmocked_cli_5 upload=metadata
+lightly-upload token=$TOKEN input_dir=$INPUT_DIR new_dataset_name=test_unmocked_cli_5 upload=metadata
 
 echo "############################### Test 6"
-lightly-upload token=${TOKEN} input_dir=${INPUT_DIR} new_dataset_name=test_unmocked_cli_6 upload=thumbnails
+lightly-upload token=$TOKEN input_dir=$INPUT_DIR new_dataset_name=test_unmocked_cli_6 upload=thumbnails
 
 echo "############################### Deleting all datasets again"
 NUMBER_OF_DATASETS=6
```

Comment:
maybe init this variable before your tests and `++` them after each command that creates a dataset and use it to create the name. This could prevent orphans created by a hastily added test

Reply:
Put then there is the risk of orphans because of forgetting to add the increment when hastily adding tests. Thus I would keep it like this. However, I added a comment telling to increase the NUMBER_OF_DATASETS if you add more tests.

Reply:
IMHO its less of a risk as one would copy/orient themselves at an already written test

Reply:
I did that.